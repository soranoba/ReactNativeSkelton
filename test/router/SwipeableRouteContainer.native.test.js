import React from 'react';
import renderer from 'react-test-renderer';
import SwipeableRouteContainer from '/router/SwipeableRouteContainer.native';
import { StaticRouter, Route } from 'react-router';
import { View } from 'react-native';

describe('SwipeableRouteContainer', () => {
  const ComponentA = () => <View style={{ backgroundColor: 'red' }} />;
  const ComponentB = () => <View style={{ backgroundColor: 'blue' }} />;
  const ComponentC = () => <View style={{ backgroundColor: 'green' }} />;

  const JSX = location => (
    <StaticRouter location={location} context={{}}>
      <SwipeableRouteContainer>
        <Route path="/a" component={ComponentA} />
        <Route path="/b" component={ComponentB} />
        <Route path="/c" component={ComponentC} />
      </SwipeableRouteContainer>
    </StaticRouter>
  );

  it('renders currectly', () => {
    const rendered = renderer.create(JSX('/a')).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders currectly, after onLayout called', () => {
    const wrapper = renderer.create(JSX('/a'));
    wrapper
      .toJSON()
      .props.onLayout({ nativeEvent: { layout: { x: 0, y: 22, width: 200, height: 400 } } });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders correctly, after transitioning from `/a` to `/c`', () => {
    const wrapper = renderer.create(JSX('/a'));
    wrapper
      .toJSON()
      .props.onLayout({ nativeEvent: { layout: { x: 0, y: 22, width: 200, height: 400 } } });
    wrapper.update(JSX('/c'));
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders correctly, after transitioning from `/c` to `/a`', () => {
    const wrapper = renderer.create(JSX('/c'));
    wrapper
      .toJSON()
      .props.onLayout({ nativeEvent: { layout: { x: 0, y: 22, width: 200, height: 400 } } });
    wrapper.update(JSX('/a'));
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
