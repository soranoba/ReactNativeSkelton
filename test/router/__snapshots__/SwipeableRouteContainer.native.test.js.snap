// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeableRouteContainer renders correctly, after transitioning from \`/a\` to \`/c\` 1`] = `
<View
  onLayout={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 400,
        "transform": Array [
          Object {
            "translateX": -200,
          },
        ],
        "width": 600,
      }
    }
  >
    <View
      style={
        Object {
          "width": 200,
        }
      }
    />
    <View
      style={
        Object {
          "width": 200,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": 200,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`SwipeableRouteContainer renders correctly, after transitioning from \`/c\` to \`/a\` 1`] = `
<View
  onLayout={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 400,
        "transform": Array [
          Object {
            "translateX": -200,
          },
        ],
        "width": 600,
      }
    }
  >
    <View
      style={
        Object {
          "width": 200,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": 200,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": 200,
        }
      }
    />
  </View>
</View>
`;

exports[`SwipeableRouteContainer renders currectly 1`] = `
<View
  onLayout={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 0,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": 0,
      }
    }
  >
    <View
      style={
        Object {
          "width": 0,
        }
      }
    />
    <View
      style={
        Object {
          "width": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`SwipeableRouteContainer renders currectly, after onLayout called 1`] = `
<View
  onLayout={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 400,
        "transform": Array [
          Object {
            "translateX": -200,
          },
        ],
        "width": 600,
      }
    }
  >
    <View
      style={
        Object {
          "width": 200,
        }
      }
    />
    <View
      style={
        Object {
          "width": 200,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": 200,
        }
      }
    />
  </View>
</View>
`;
